volumes/vds/sql/select/select-row_count_lookback.sql

```diff
@@ -5,31 +5,33 @@
 WITH lookback AS ( --noqa: L045
     SELECT
         date_trunc('day', {{ params.dt_col }}) AS _dt, --noqa: L039
-        COUNT(*) AS lookback_count
+        SUM({{ params.col_to_sum }}) AS lookback_count
     FROM {{ params.table }}
     WHERE
         {{ params.dt_col }} >= '{{ ds }} 00:00:00'::timestamp - interval '{{ params.lookback }}'
         AND {{ params.dt_col }} < '{{ ds }} 00:00:00'::timestamp
-        AND division_id = {{ params.div_id }}::int
     --group by day then avg excludes missing days.
     GROUP BY _dt --noqa: L003
+),
+
+today AS (
+    SELECT
+        SUM({{ params.col_to_sum }}) AS today_count
+    FROM {{ params.table }}
+    WHERE
+        {{ params.dt_col }} >= '{{ ds }} 00:00:00'::timestamp
+        AND {{ params.dt_col }} < '{{ ds }} 00:00:00'::timestamp + interval '1 day'
 )
 
 SELECT
-    COUNT(*) >= FLOOR({{ params.threshold }}::numeric * lb.lookback_avg) AS check, 
-    'Daily count: ' || to_char(COUNT(*), 'FM9,999,999,999') AS ds_count,
+    a.today_count >= FLOOR({{ params.threshold }}::numeric * AVG(lb.lookback_count)) AS check, 
+    'Daily count: ' || to_char(a.today_count, 'FM9,999,999,999') AS ds_count,
     initcap('{{ params.lookback }}') || ' Lookback Avg: '
-        || to_char(lb.lookback_avg, 'FM9,999,999,999') AS lookback_avg,
+        || to_char(AVG(lb.lookback_count), 'FM9,999,999,999') AS lookback_avg,
     'Pass threshold: ' || to_char(
-            FLOOR({{ params.threshold }}::numeric * lb.lookback_avg),
+            FLOOR({{ params.threshold }}::numeric * AVG(lb.lookback_count)),
             'FM9,999,999,999'
             ) AS passing_value
-FROM {{ params.table }} AS a,
-    LATERAL (
-        SELECT AVG(lookback_count) lookback_avg FROM lookback
-    ) AS lb
-WHERE
-    a.{{ params.dt_col }} >= '{{ ds }} 00:00:00'::timestamp
-    AND a.{{ params.dt_col }} < '{{ ds }} 00:00:00'::timestamp + interval '1 day'
-    AND a.division_id = {{ params.div_id }}::int
-GROUP BY lb.lookback_avg
\ No newline at end of file
+FROM today AS a,
+    lookback AS lb
```

Comment:
`lb` is used in line 32. From where did this warning come? Have you tested the code? Could it be misspelled above, e.g., `1` instead of `l`?